autotest/pyscripts/test_gdal_utils_retcodes.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env pytest
+# -*- coding: utf-8 -*-
+###############################################################################
+#
+# Project:  GDAL/OGR Test Suite
+# Purpose:  osgeo_utils (gdal-utils) testing
+# Author:   Matt Wilkie <maphew@gmail.ca>
+#
+###############################################################################
+# Copyright (c) 2022, Matt Wilkie <maphew@gmail.ca>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+"""
+Verify that gdal-utils scripts all exit with the same return code when
+called without arguments.
+
+Spawned from https://github.com/OSGeo/gdal/issues/5561
+
+Programs: python files installed into PYTHONHOME/Scripts and can be executed
+as just names e.g. 'gdal_edit somefile.tif ...'
+
+Scripts: python files that need to called from python in order to be run,
+e.g. 'python path/to/samples/pct2rgb.py somefile.tif ...'
+"""
+import glob
+import sys
+import subprocess
+import pytest
+from pathlib import Path
+
+# pytest.skip("THIS TEST IN DRAFT MODE, SKIPPING", allow_module_level=True)
+
+# here = r"D:\code\public\gdal\autotest\pyscripts"
+# script_path = r"D:\code\public\gdal\swig\python\gdal-utils\osgeo_utils"
+here = Path(__file__).parent.absolute()
+script_path = Path(here / "../../swig/python/gdal-utils/osgeo_utils/").resolve()
+
+excludes = [
+    "setup.py",
+    "__init__.py",
+    "gdal_auth.py",  # gdal_auth doesn't take arguments
+]
+
+
+def get_scripts(script_path, excludes):
+    # Using glob instead of Path.glob because removing excludes from a list of
+    # strings is easier than with objects
+
+    # scripts = list(script_path.glob("**/.py" ))
+    s1 = glob.glob("*.py", root_dir=script_path)
```

Comment:
root_dir is a Python 3.10 addition. Our minimum version is 3.6 currently

Reply:
Thanks!